From: Marc Kleine-Budde <m.kleine-budde@pengutronix.de>
Subject: build mkg3states for build not for host

crosscompiling sux, x86 doesn't like it, when arm binaries are executed:

/bin/sh ../libtool --mode=link arm-softfloat-linux-gnu-gcc -O
-L/home/frogger/projects/pengutronix/ptxdist/ptxdist-0.9-trunk/local/arm-softfloat-linux-gnu/lib
-L/home/frogger/projects/pengutronix/ptxdist/ptxdist-0.9-trunk/local/arm-softfloat-linux-gnu/usr/lib
-o mkg3states mkg3states.o arm-softfloat-linux-gnu-gcc -O -o
mkg3states mkg3states.o
-L/home/frogger/projects/pengutronix/ptxdist/ptxdist-0.9-trunk/local/arm-softfloat-linux-gnu/lib
-L/home/frogger/projects/pengutronix/ptxdist/ptxdist-0.9-trunk/local/arm-softfloat-linux-gnu/usr/lib
rm -f g3states.h ./mkg3states -c > g3states.h_ && mv g3states.h_
g3states.h /bin/sh: ./mkg3states: cannot execute binary file
make[3]: *** [g3states.h] Error 126

Signed-off-by: Marc Kleine-Budde <m.kleine-budde@pengutronix.de>
---

# 20060619 lfu: rediffed for Xorg7.1
# 20060119 mkl: submitted upstream

Index: liblbxutil-1.0.1/configure.ac
===================================================================
--- liblbxutil-1.0.1.orig/configure.ac	2007-10-18 09:39:19.000000000 +0200
+++ liblbxutil-1.0.1/configure.ac	2007-10-18 09:40:53.000000000 +0200
@@ -51,4 +51,5 @@
 
 AC_OUTPUT([Makefile
 	   src/Makefile
+	   src/image/Makefile
 	   lbxutil.pc])
Index: liblbxutil-1.0.1/src/Makefile.am
===================================================================
--- liblbxutil-1.0.1.orig/src/Makefile.am	2007-10-18 09:39:19.000000000 +0200
+++ liblbxutil-1.0.1/src/Makefile.am	2007-10-18 09:39:22.000000000 +0200
@@ -1,9 +1,6 @@
-lib_LTLIBRARIES = liblbxutil.la
-
-noinst_PROGRAMS = mkg3states
+SUBDIRS = image
 
-mkg3states_SOURCES =				\
-	$(srcdir)/image/mkg3states.c
+lib_LTLIBRARIES = liblbxutil.la
 
 liblbxutil_la_SOURCES =				\
 	$(srcdir)/lbx_zlib/reqstats.h		\
@@ -28,9 +25,9 @@
 
 $(srcdir)/image/dfaxg42d.c: g3states.h
 
-g3states.h: mkg3states
+g3states.h: image/mkg3states
 	-rm -f g3states.h
-	./mkg3states -c > g3states.h_ && mv g3states.h_ g3states.h
+	./image/mkg3states -c > g3states.h_ && mv g3states.h_ g3states.h
 
 liblbxutil_la_LDFLAGS = -version-number 1:0:0 -no-undefined
 
Index: liblbxutil-1.0.1/src/image/Makefile.am
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ liblbxutil-1.0.1/src/image/Makefile.am	2007-10-18 09:39:27.000000000 +0200
@@ -0,0 +1,15 @@
+# evil hack
+CFLAGS=$(CFLAGS_FOR_BUILD)
+CPPFLAGS=$(CPPFLAGS_FOR_BUILD)
+LDFLAGS=$(LDFLAGS_FOR_BUILD)
+
+CC=$(CC_FOR_BUILD)
+LIBTOOL = @LIBTOOL@ --tag=CC
+
+noinst_PROGRAMS = mkg3states
+
+mkg3states_SOURCES =				\
+	mkg3states.c
+
+mkg3states_CFLAGS=$(CFLAGS_FOR_BUILD)
+mkg3states_LDFLAGS=$(LDFLAGS_FOR_BUILD)
